src/BackgroundLayerChooser/BackgroundLayerChooser.spec.tsx

```diff
@@ -0,0 +1,189 @@
+    });
+
+    it('can be rendered', () => {
+      const { container } = renderInMapContext(
+        map,
+        <BackgroundLayerChooser
+          layers={layers}
+          allowEmptyBackground={true}
+        />
+      );
+      expect(container.querySelector('.bg-layer-chooser')).toBeInTheDocument();
+    });
+  });
+
+  it('shows layer cards when button is clicked', async () => {
+    const { container } = renderInMapContext(
+      map,
+      <BackgroundLayerChooser
+        layers={layers}
+        allowEmptyBackground={true}
+      />
+    );
+    const btn = container.querySelector('.change-bg-btn');
+    await act(async () => {
+      btn && fireEvent.click(btn);
+    });
+    await waitFor(() => {
+      expect(container.querySelector('.layer-cards')).toBeInTheDocument();
+      expect(container.querySelectorAll('.layer-preview').length).toBeGreaterThan(0);
+    });
+  });
+
+  it('calls titleRenderer for each layer', async () => {
+    const titleRenderer = jest.fn(l => <span>Layer: {l.get('name')}</span>);
+    const { container } = renderInMapContext(
+      map,
+      <BackgroundLayerChooser
+        layers={layers}
+        allowEmptyBackground={true}
+        titleRenderer={titleRenderer}
+      />
+    );
+    const btn = container.querySelector('.change-bg-btn');
+    await act(async () => {
+      btn && fireEvent.click(btn);
+    });
+    await waitFor(() => {
+      expect(titleRenderer).toHaveBeenCalled();
+    });
+  });
+
+  it('shows no background option and selects it', async () => {
+    const { container } = renderInMapContext(
+      map,
+      <BackgroundLayerChooser
+        layers={layers}
+        allowEmptyBackground={true}
+        noBackgroundTitle="None"
+      />
+    );
+    const btn = container.querySelector('.change-bg-btn');
+    await act(async () => {
+      btn && fireEvent.click(btn);
+    });
+    const noBg = await waitFor(() => container.querySelector('.no-background'));
+    expect(noBg).toBeInTheDocument();
+    await act(async () => {
+      noBg && fireEvent.click(noBg);
+    });
+    await waitFor(() => {
+      expect(container.querySelector('.bg-preview .layer-title')?.textContent).toBe('None');
+    });
+  });
+
+  it('selects a background layer when preview is clicked', async () => {
+    const { container } = renderInMapContext(
+      map,
+      <BackgroundLayerChooser
+        layers={layers}
+        allowEmptyBackground={true}
+      />
+    );
+    const btn = container.querySelector('.change-bg-btn');
+    await act(async () => {
+      btn && fireEvent.click(btn);
+    });
+    const previews = await waitFor(() => container.querySelectorAll('.layer-preview'));
+    expect(previews.length).toBeGreaterThan(0);
+    await act(async () => {
+      previews[0] && fireEvent.click(previews[0]);
+    });
+    await waitFor(() => {
+      expect(container.querySelector('.bg-preview .layer-title')?.textContent).toBe('OSM');
```

Comment:
```suggestion
      expect(container.querySelector('.bg-preview .layer-title')?.textContent).toBe(layers[0].get('name'));
```
don't know if this is really better ...